File: editor/en/[645]Set Mismatch.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function (nums) {
    let set = new Set();
    let dup = 0;
    let miss = 0;
    let sum = 0;

    for (let num of nums) {
        if (set.has(num)) {
            dup = num;
        }

        set.add(num);
        sum += num;
    }

    miss = (nums.length) * (nums.length + 1) / 2 - sum + dup;
    return [dup, miss];
};
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[645]Set Mismatch.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int[] findErrorNums(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int dup = 0;
        int miss = 0;
        int sum = 0;

        for (int num : nums) {
            if (!set.add(num)) {
                dup = num;
            }

            sum += num;
        }

        miss = (nums.length) * (nums.length + 1) / 2 - sum + dup;
        return new int[]{dup, miss};
    }
}
//leetcode submit region end(Prohibit modification and deletion)
